hardware_interface/src/lexical_casts.cpp

```diff
@@ -28,8 +32,28 @@ double stod(const std::string & s)
     throw std::invalid_argument("Failed converting string to real number");
```

Comment:
You should return `std::nullopt` here. The top-level `stod` function will throw the exception for you if you return an empty optional